unaligned 64-bit atomic operation on i386

##### Description

I've recently [packaged this library for Debian](https://tracker.debian.org/pkg/golang-github-ibm-sarama), as it is required to upgrade another library. When the tests were run on a 32bit system (i386), the following panic is reported:

```
=== RUN   TestConsumerHandlesExpectationsPausingResuming
    consumer.go:284: Expectations set on other/0, but no partition consumer was started.
    consumer.go:284: Expectations set on test/1, but no partition consumer was started.
    consumer.go:284: Expectations set on test/0, but no partition consumer was started.
--- FAIL: TestConsumerHandlesExpectationsPausingResuming (0.00s)
panic: unaligned 64-bit atomic operation [recovered]
	panic: unaligned 64-bit atomic operation

goroutine 49 [running]:
testing.tRunner.func1.2({0x81c13e0, 0x822cc58})
	/usr/lib/go-1.22/src/testing/testing.go:1631 +0x2a3
testing.tRunner.func1()
	/usr/lib/go-1.22/src/testing/testing.go:1634 +0x434
panic({0x81c13e0, 0x822cc58})
	/usr/lib/go-1.22/src/runtime/panic.go:770 +0x103
runtime/internal/atomic.panicUnaligned()
	/usr/lib/go-1.22/src/runtime/internal/atomic/unaligned.go:8 +0x2d
runtime/internal/atomic.Xadd64(0xa49e944, 0x1)
	/usr/lib/go-1.22/src/runtime/internal/atomic/atomic_386.s:125 +0x11
github.com/IBM/sarama/mocks.(*PartitionConsumer).YieldMessage(0xa49e910, 0xa4c8ea0)
	/tmp/autopkgtest-lxc.3ncu_7nv/downtmp/autopkgtest_tmp/_build/src/github.com/IBM/sarama/mocks/consumer.go:400 +0xcb
github.com/IBM/sarama/mocks.TestConsumerHandlesExpectationsPausingResuming(0xa4eec08)
	/tmp/autopkgtest-lxc.3ncu_7nv/downtmp/autopkgtest_tmp/_build/src/github.com/IBM/sarama/mocks/consumer_test.go:82 +0x1e2
testing.tRunner(0xa4eec08, 0x81ff324)
	/usr/lib/go-1.22/src/testing/testing.go:1689 +0x125
created by testing.(*T).Run in goroutine 1
	/usr/lib/go-1.22/src/testing/testing.go:1742 +0x3b9
FAIL	github.com/IBM/sarama/mocks	0.008s
```

##### Versions

<!-- Please specify real version numbers or git SHAs, not just "Latest" since that changes fairly regularly. -->

| Sarama | Kafka | Go  |
| ------ | ----- | --- |
|  1.43.0 and 1.43.1      |  N/A    |   1.22.2  |

##### Configuration

<!-- What configuration values are you using for Sarama and Kafka? -->

Debian sid, with go 1.22.2 on an i386 system (both the auto test builder and a local container where I reproduced the problem). I don't know if this affects armel/armhf, or if this is specific to the 32bit i386 architecture.

##### Logs

Full log available [here](https://ci.debian.net/packages/g/golang-github-ibm-sarama/testing/i386/45551928/).

##### Additional Context

<!-- Add any other context about the problem here. -->

I've taken a quick look for an easy fix like an unaligned struct field, but didn't see anything obvious.

--- a/.github/workflows/i386.yml
+++ b/.github/workflows/i386.yml
@@ -1,0 +1,39 @@
+name: i386
+on:
+  merge_group:
+  push:
+    branches:
+    - main
+    paths-ignore:
+    - '**/*.md'
+  pull_request:
+    branches:
+    - "**"
+    paths-ignore:
+    - '**/*.md'
+
+permissions:
+  contents: read  # for actions/checkout to fetch code
+
+jobs:
+  atomicalign:
+    permissions:
+      contents: read  # for actions/checkout to fetch code
+      pull-requests: read  # for golangci/golangci-lint-action to fetch pull requests
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        persist-credentials: false
+    - name: Setup Go
+      uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+      with:
+        go-version: 1.22.x
+    - name: staticcheck
+      env:
+        GOARCH: 386
+        GOFLAGS: -tags=functional
+      run: |
+          git clone --depth=1 https://github.com/dominikh/go-tools /tmp/go-tools
+          ( cd /tmp/go-tools/cmd/staticcheck && go build -o /tmp/staticcheck )
+          /tmp/staticcheck -checks SA1027 ./...

--- a/mocks/consumer.go
+++ b/mocks/consumer.go
@@ -248,6 +248,7 @@ func (c *Consumer) ExpectConsumePartition(topic string, partition int32, offset 
 // channels using YieldMessage and YieldError.
 type PartitionConsumer struct {
 	highWaterMarkOffset           int64 // must be at the top of the struct because https://golang.org/pkg/sync/atomic/#pkg-note-BUG
+	suppressedHighWaterMarkOffset int64
 	l                             sync.Mutex
 	t                             ErrorReporter
 	topic                         string
@@ -255,7 +256,6 @@ type PartitionConsumer struct {
 	offset                        int64
 	messages                      chan *sarama.ConsumerMessage
 	suppressedMessages            chan *sarama.ConsumerMessage
-	suppressedHighWaterMarkOffset int64
 	errors                        chan *sarama.ConsumerError
 	singleClose                   sync.Once
 	consumed                      bool

